miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ContainerRestController.java

```diff
@@ -147,4 +150,53 @@ public void bulkDelete(@RequestBody(required = true) List<Long> ids) throws IOEx
     return MisoWebUtils.generateSpreadsheet(input, PartitionSpreadsheets::valueOf, request, response);
   }
 
+  private static class SerialNumberValidationDto {
+    private final String serialNumber;
+    private final String containerId;
+
+    public SerialNumberValidationDto(String serialNumber, String containerId) {
+      this.serialNumber = serialNumber;
+      this.containerId = containerId;
+    }
+
+    public String getSerialNumber() {
+      return serialNumber;
+    }
+
+    public String getContainerId() {
+      return containerId;
+    }
+  }
+
+  @PostMapping(value = "/validate-serial-number")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public void validateSerialNumber(@RequestBody SerialNumberValidationDto params, HttpServletRequest request, HttpServletResponse response,
+      UriComponentsBuilder uriBuilder) throws IOException {
+    String serialNumber = params.getSerialNumber();
+    String maybeContainerId = params.getContainerId();
+
+    if (isStringEmptyOrNull(serialNumber) || isStringEmptyOrNull(maybeContainerId)) {
+      throw new RestException("Serial number and containerID must be provided", Status.BAD_REQUEST);
+    }
+
+    Long containerId = SequencerPartitionContainerImpl.UNSAVED_ID;
+    if (!maybeContainerId.contains("Unsaved")) {
+      containerId = Long.valueOf(maybeContainerId);
+    }
+    try {
+      List<SequencerPartitionContainer> matchingContainers = (List<SequencerPartitionContainer>) containerService
+          .listByBarcode(serialNumber);
+      if (matchingContainers.isEmpty()
+          || (matchingContainers.size() == 1 && matchingContainers.get(0).getId() == containerId)) {
+        return;
+      } else {
+        throw new RestException("Serial number is already associated with another container", Status.PRECONDITION_FAILED);
```

Comment:
All instances of `Status.PRECONDITION_FAILED` should be changed to `Status.BAD_REQUEST`. Precondition failed status refers to conditions specified in request headers - not the user/application data